lib/plugin/instance.rb

```diff
@@ -1347,6 +1347,24 @@ def add_request_rate_limiter(
     end
   end
 
+  # This method allows plugins to preload topic associations when loading topics
+  # that make use of topic_list.
+  #
+  # @param fields [Array<Symbol>, Symbol, Hash] The topic associations to preload.
+  #   - :association` [Symbol] The association to preload.
+  #   - :condition` [Proc] A function that returns a boolean value indicating whether the association should be preloaded
+  #
+  # @example
+  #   register_topic_preloader_associations(%i[custom_association another_association])
+  #   register_topic_preloader_associations({
+  #     association: :linked_topic, condition: ->(topic) { topic.custom_field.present? }
+  #   })
```

Comment:
I think a better pattern may be to yield the `topic_preloader_associations` to a block is registered so that the fields params of this API must be of a single type rather than 3 possible types.

```
register_topic_preloader_associations do |topic_preloader_associations|
  if condition
    topic_preloader_associations.push(....)
  end
end
```